### DISCUSSION TITLE
Style of map

### DISCUSSION THREAD

**Discussion by alainlietard** [freeplane/freeplane#850] [Help]
Hello 
I would like to use the style of this document 
Than you for your help
Best regards
Alain 
![70CC5B10-BF52-4577-A7FA-2002FBFE019E](https://user-images.githubusercontent.com/111346725/204080380-833bfbe1-c980-4c75-aefa-c6cdd6dfa7e2.jpeg)

**Comment by euu2021** [comment]
I guess your difficulty is in getting the nodes to alternate color automatically, on each level, right?

**Comment by EdoFro** [comment]
@alainlietard
Can you attach a mindmap or textfile with the data, to use it to create a same map?

**Comment by EdoFro** [comment]
sample

**Comment by EdoFro** [comment]
@alainlietard 
ok. I created a sample map with styles for your use case

it looks like this:
![image](https://user-images.githubusercontent.com/35700575/204100936-e50e2049-8648-413c-9d60-c60579c4d709.png)

### custom styles
it has 5 custom styles:

![image](https://user-images.githubusercontent.com/35700575/204101114-78298bb2-e6d9-477a-aa46-0b8ff03e3501.png)

and their explanation:

![image](https://user-images.githubusercontent.com/35700575/204101488-df1cfcd6-5dde-45da-bb90-516fc9542c4e.png)

### conditional styles rules

#### 1. white
  - it uses a script filter
  - rule:
    - it applies only to "move" styled nodes
    - it counts the "move" styled nodes in its way to the root and if its an odd number it paint it "white"
  - exits conditional formating

#### 2. black
  - Only if the white rule was not applied to the node before
  - rule:
    - it applies only to "move" styled nodes

the script filter from white is:

```groovy
if(node.style.name!='move') return false
node.pathToRoot.count{n-> n.style.name=='move'}%2==1
```

### turn nodes formula

the turn nodes have also a formula to automatically increase their number

formula
```groovy
=
def t = node.parent.displayedText.takeBefore('.')
t = t.isNumber()? t.toInteger() : 0
return "${t + 1}."
```
here it is the map:
[Chess template.zip](https://github.com/freeplane/freeplane/files/10096746/Chess.template.zip)


hope it helps,

edo

**Comment by alainlietard** [comment]
Many thanks !!!

Wonderful !!!

Best regards

**Reply by EdoFro** [reply]
@alainlietard,
May you post a final image of your map (or a branch of it) to see how it looks? 
Maybe it can be used  as an additional example of the things that can be made with Freeplane.

Thanks,
edo

**Comment by alainlietard** [comment]
Hello

I'm using this and I got many problems about formation.

I'm thinking about xmind

**Reply by EdoFro** [reply]
Hi @alainlietard,
I would like to hear which problems did you have.
It's difficult to improve if I can't get any feedback.

bye,
